### PR TITLE
fix(core): Do not print errors prop for non-AggregateError errors

### PR DESCRIPTION
**Pull request by kamilogorek**
As mentioned in the comment, there is currently no way to detect `AggregateError` via `rusty_v8`.
We can migrate to it once v8 itself exposes `v8__Exception__AggregateError`, and we add binding for it.

Fixes https://github.com/denoland/deno/issues/16340

### CODE REVIEW

**Comment by bartlomieju** [comment]
> As mentioned in the comment, there is currently no way to detect `AggregateError` via `rusty_v8`. We can migrate to it once v8 itself exposes `v8__Exception__AggregateError`, and we add binding for it.
> 
> Fixes #16340

I don't see `AggregateError` being exposed publicly in V8 header files, where did you get that information?

**Comment by kamilogorek** [comment]
This is the thing, that it's not, so we have to work around it for now :)

I'm debugging test panic now, should get it fixed soon.

**Comment by kamilogorek** [comment]
Updated to a more generic solution and fixed tests (CI pending). I'm quite new to the codebase, so if there is a better solution available somewhere in the helpers, just let me know and I'll update the PR.

**Comment by kamilogorek** [comment]
Calling `(new DOMException).__proto__.name` throws `Uncaught TypeError: Illegal invocation`, and for some reason `is_instance_of_error(scope, prototype)` is letting this go through, despite the fact that `DOMException instanceof Error === false`.

**Comment by nayeemrmn** [comment]
> Calling `(new DOMException).__proto__.name` throws `Uncaught TypeError: Illegal invocation`

That's browser behaviour, seems fine

> `is_instance_of_error(scope, prototype)` is letting this go through, despite the fact that `DOMException instanceof Error === false`

I think you meant to do `new DOMException() instanceof Error` (`true`)

**Comment by kamilogorek** [comment]
> That's browser behaviour, seems fine

Agree, it's just that I didn't find an easy way to prevent the check here.

> I think you meant to do `new DOMException() instanceof Error` (true)

Right :) (although the result is the same in this case)

Found another way to check for what we need, and all tests appear to be passing now.
Is there maybe a better, already existing way of asking for a `Value` constructor?

**Comment by kamilogorek** [comment]
We can also skip some checks and use `v8::Object` directly with:

<details>
  <summary>Patch</summary>

```patch
diff --git core/error.rs core/error.rs
index 870f0bab2..e9d1403d8 100644
--- core/error.rs
+++ core/error.rs
@@ -325,7 +325,6 @@ impl JsError {
     }
 
     if is_instance_of_error(scope, exception) {
-      let v8_exception = exception;
       // The exception is a JS Error object.
       let exception: v8::Local<v8::Object> = exception.try_into().unwrap();
       let cause = get_property(scope, exception, "cause");
@@ -445,7 +444,7 @@ impl JsError {
       }
 
       let mut aggregated: Option<Vec<JsError>> = None;
-      if is_aggregate_error(scope, v8_exception) {
+      if is_aggregate_error(scope, exception) {
         // Read an array of stored errors, this is only defined for `AggregateError`
         let aggregated_errors = get_property(scope, exception, "errors");
         let aggregated_errors: Option<v8::Local<v8::Array>> =
@@ -584,15 +583,10 @@ pub(crate) fn is_instance_of_error<'s>(
 /// and we cannot create bindings for it. This forces us to rely on `name` inference.
 pub(crate) fn is_aggregate_error<'s>(
   scope: &mut v8::HandleScope<'s>,
-  value: v8::Local<v8::Value>,
+  value: v8::Local<v8::Object>,
 ) -> bool {
   let mut maybe_prototype = Some(value);
   while let Some(prototype) = maybe_prototype {
-    if !prototype.is_object() {
-      return false;
-    }
-
-    let prototype = prototype.to_object(scope).unwrap();
     let prototype_name = match get_property(scope, prototype, "constructor") {
       Some(constructor) => {
         let ctor = constructor.to_object(scope).unwrap();
@@ -605,7 +599,13 @@ pub(crate) fn is_aggregate_error<'s>(
       return true;
     }
 
-    maybe_prototype = prototype.get_prototype(scope);
+    maybe_prototype = prototype.get_prototype(scope).and_then(|p| {
+      if p.is_object() {
+        Some(p.to_object(scope).unwrap())
+      } else {
+        None
+      }
+    });
   }
 
   false
```
</details>

Both work and I don't mind either.

**Comment by bartlomieju** [comment]
@kamilogorek seems it's roughly the same amount of code - maybe let's just continue with current implementation so that we don't pollute `git blame` with many small changes?

File: cli/tests/run_tests.rs [hunk 1]
```diff
@@ -2780,6 +2780,12 @@ mod run {
     exit_code: 1,
   });
 
+  itest!(error_with_errors_prop {
```
Comment (bartlomieju) [inline cli/tests/run_tests.rs:2783]: Please add a comment here linking to relevant issue - it's not immediately obvious what we are testing here.

**Review by bartlomieju — COMMENTED** [review]
Looks good, thanks for working on this @kamilogorek! Just one small nitpick from me

**Review by bartlomieju — APPROVED** [review]
LGTM, great fix @kamilogorek, thanks!

**Pull request merged**